engine/timer.cpp

```diff
@@ -0,0 +1,50 @@
+#include "timer.h"
+
+#include <spdlog/spdlog.h>
+
+#include <algorithm>
+#include <chrono>
+#include <iostream>
+#include <memory>
+
+namespace pong
+{
+
+static constexpr int MS_TO_US = 1000;
+static constexpr int S_TO_US = 1000000;
+
+float Timer::frameTime = 1.0f;
+
+void Timer::HandleTimerCallbacks()
+{
+    const auto timeWaited = std::chrono::duration_cast<std::chrono::microseconds>(std::chrono::system_clock::now() - mLastTime);
+    mLastTime = std::chrono::system_clock::now();
+
+    Timer::frameTime = static_cast<float>(timeWaited.count()) / static_cast<float>(S_TO_US);
+
+    const int timeWaitedUs = static_cast<int>(timeWaited.count());
+    for (auto& timerRequest : mActiveTimers)
+    {
+        timerRequest->mTimeoutUs -= timeWaitedUs;
+        if (timerRequest->mTimeoutUs <= 0)
+        {
+            timerRequest->mCallback();
```

Comment:
I recognize this timer pattern, seems familiar to an old codebase I used to work in